test/integration/enumerate_databases.test.ts

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+
+import { AddUserOptions, MongoClient, MongoServerError } from '../../src';
+
+describe('listDatabases', function () {
+  // TODO(NODE3860): Create driver test variants that run with AUTH enabled
+  beforeEach(function () {
+    if (process.env.AUTH !== 'auth') {
+      this.skip();
+    }
+  });
+
+  describe('authorizedDatabases flag', function () {
```

Comment:
general question - do we not need to cover the different server versions? is 4.0.5 server not in our pipeline?

Reply:
capturing convo from slack - we shouldn't need to break out tests for before and after that version, our choice of making a user with a `read` role should give the same behavior across versions.